src/patchelf.cc

```diff
@@ -1470,11 +1485,6 @@ void ElfFile<ElfFileParamNames>::modifyRPath(RPathOp op,
     if (rpath && rpath == newRPath) {
         return;
     }
-
-    if (op == rpAdd) {
-        newRPath = std::string(rpath ? rpath : "") + ":" + newRPath; 
```

Comment:
I think this also fixes a bug where `rpAdd` would have no effect if the above condition would return early.